README.md

```diff
@@ -678,16 +680,26 @@ Some things to note about the above code:
 * A `timeout` can optionally be provided which will throw a `asyncio.TimeoutError` if reached (internally backed by
   `asyncio.wait_for` which uses a timer)
 
-#### Asyncio and Cancellation
+#### Asyncio and Determinism
 
-Workflows are backed by a custom [asyncio](https://docs.python.org/3/library/asyncio.html) event loop. This means many
-of the common `asyncio` calls work as normal. Some asyncio features are disabled such as:
+Workflows must be deterministic. Workflows are backed by a custom
+[asyncio](https://docs.python.org/3/library/asyncio.html) event loop. This means many of the common `asyncio` calls work
+as normal. Some asyncio features are disabled such as:
 
 * Thread related calls such as `to_thread()`, `run_coroutine_threadsafe()`, `loop.run_in_executor()`, etc
 * Calls that alter the event loop such as `loop.close()`, `loop.stop()`, `loop.run_forever()`,
   `loop.set_task_factory()`, etc
 * Calls that use anything external such as networking, subprocesses, disk IO, etc
 
+Also, there are some `asyncio` utilities that internally use `set()` which can make them non-deterministic from one
+worker to the next. Therefore the following `asyncio` functions have `workflow`-module alternatives that are
+deterministic:
+
+* `asyncio.as_completed()` - use `workflow.as_completed()`
+* `asyncio.wait()` - use `workflow.wait()`
```

Comment:
I'm wondering about naming these `workflow.asyncio_wait()` and`workflow.asyncio_as_completed()`.

The main reason is that `workflow.wait()` will be confused with `workflow.wait_condition()`. It may just be a hiccup that they'll get over, but it would be nice not to cause users that hiccup.

Another reason is that `workflow.asyncio_xxx()` is just directly saying exactly what it is: a shim that is basically exactly the same as the utility from `asyncio`. It makes it clear that they are utilities for dealing with arbitrary tasks/coroutines rather than being a special sort of `as_completed` or `wait` that has to do with waiting or completing Temporal-specific entities.